plugins/woocommerce/client/blocks/tests/e2e/tests/product-collection/product-collection-errors.block_theme.spec.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { Page } from '@playwright/test';
+import { test, expect, Editor, wpCLI } from '@woocommerce/e2e-utils';
+
+test.describe( 'Product Page: error notices when adding out-of-stock products', () => {
+	test( 'displays error notice when attempting to add product beyond stock limit', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		const productName = 'A Managed Stock';
+
+		wpCLI(
+			`wc product create --name="${ productName }" --regular_price=10 --manage_stock=true --stock_quantity=1 --user=admin`
+		);
+		await admin.createNewPost();
+		await editor.insertBlock( { name: 'woocommerce/product-collection' } );
+
+		const singleProduct = await editor.getBlockByName(
+			'woocommerce/product-collection'
+		);
+
+		await expect(
+			singleProduct.getByText( 'create your own' )
+		).toBeVisible();
+
+		await singleProduct.getByText( 'create your own' ).click();
+
+		await editor.publishAndVisitPost();
```

Comment:
You can replace this with:

```suggestion
		await pageObject.createNewPostAndInsertBlock();
		await pageObject.publishAndGoToFrontend();
```